server/src/server.ts

```diff
@@ -425,6 +407,49 @@ connection.onCompletion((pos: CompletionParams): CompletionItem[] => {
     return completionProvider.getAvailableCompletionItems(pos.textDocument.uri, doc.offsetAt(pos.position));
 });
 
+connection.onDocumentFormatting((_params) => {
+    const doc = documents.get(_params.textDocument.uri);
+    // We should directly send an empty list if this feature is disabled by the settings.
+    return buildDocumentEditList(doc, SETTINGS.documentFormat);
+});
+
+/**
+ * Build and send the list of Text Edit to apply to `document`
+ * accordingly to the settings defined in `documentFormatSettings`.
+ *
+ * @param document the text document to format
+ * @param documentFormatSettings the document format settings to apply
+ */
+export function buildDocumentEditList(document: TextDocument, documentFormatSettings?: IDocumentFormatSettings) {
+    const inputStream = CharStreams.fromString(document.getText(), document.uri);
+    const lexer = new YmlLexer(inputStream);
+    const tokenStream = new CommonTokenStream(lexer);
+    const parser = new YmlParser(tokenStream);
+    // No need to syntax errors to the client.
+    parser.removeErrorListeners();
+
+    // Parse the input.
+    const result = parser.kaoFile();
+    const edits: TextEdit[] = [];
+    const visitor = new YmlKaoFileVisitor(
+        new YmlCompletionItemsProvider(),
+        document.uri,
+        new YmlDefinitionProvider(),
```

Comment:
Idem? Useless here? Could be replaced with an idle or dummy object or even null? Not sure how that works. Also dunno how costly it is to create this. But gosh we do so for every formatted file?

Reply:
I'm not sure we have correct null-checks, so I prefer leaving it here.
I'll make it cleaner later.